crates/ide_assists/src/handlers/remove_unused_param.rs

```diff
@@ -315,30 +350,23 @@ fn bar() {
     }
 
     #[test]
-    fn remove_method_param() {
-        // FIXME: This is completely wrong:
-        //  * method call expressions are not handled
-        //  * assoc function syntax removes the wrong argument.
+    fn test_remove_method_param() {
         check_assist(
             remove_unused_param,
             r#"
 struct S;
 impl S { fn f(&self, $0_unused: i32) {} }
 fn main() {
     S.f(92);
-    S.f();
-    S.f(92, 92);
```

Comment:
Let's leave this in -- it's important to check that the code doesn't crash even if there are some type errors.

Reply:
So when I add these and run the tests, it works for the first case, as there's no arguments and it has nothing to do, but for the second case, i.e. S.f(92, 92), it removes the first argument, as it should normally do with correct cases.